### ISSUE TITLE
Beam does not stop working after a runtime exception.

### ISSUE THREAD

**Issue by nikolayilyin** [LBNL-UCB-STI/beam#3524]
In beam config there is a key `beam.output.writePlansAndStopSimulation`. 
If this key set to true BEAM should write generated plans and stop the simulation with runtime exception, but, after runtime exception the simulation continue to hang.
Probably something is not stopped even when runtime exception happens, it needs to be investigated and fixed.

**Comment by g-derzhavets** [comment]
I did some research on the problem. I've tried to throw the runtime exception in different parts of the system and by doing that, I discovered, that if you throw the exception before [this injector is created](https://github.com/LBNL-UCB-STI/beam/blob/e5a603fab1b0e6b78cdb13321095f090e90da388/src/main/scala/beam/sim/BeamHelper.scala#L642) then the processes finish successfully. If you throw after - the processes hang. 
I've also tried to do the same experiment on commit from July 2020 and there is the same behaviour.
```
commit cf5b1cd7f1e32df5b8250b33371b584f9b4453d2 (HEAD -> develop)
Merge: 1774e253e 8d4e11fd3
Author: Artavazd Balaian <reasyu@gmail.com>
Date:   Thu Jul 9 22:24:06 2020 +0700
```


From what I understand this happens because of the [code around here ](https://github.com/LBNL-UCB-STI/matsim/blob/0.10.x/matsim/src/main/java/org/matsim/core/controler/MatsimRuntimeModifications.java#L85). Please also see this comment 

			// Propagate Exception in case Controler.run is called by someone who wants to catch
			// it. It is probably not strictly correct to wrap the exception here.
			// But otherwise, this method would have to declare "throws Throwable".
			// In theory, a run method for test cases would probably need to
			// be different from the run method of the "MATSim platform" which
			// takes control of the JVM by installing hooks and exception
			// handlers.

**Comment by g-derzhavets** [comment]
The core reason of the application hanging is a lock in thread, related to the `LoggingEventsManager`, somewhere around this code https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/scala/beam/sim/LoggingEventsManager.scala#L122

![image](https://user-images.githubusercontent.com/98255053/180992179-4de9122c-6c0e-4dfa-9e03-1208c83e4ee4.png)


If I kill this thread explicitly - the application exits correctly.

**Comment by nikolayilyin** [comment]
Fixed by #3566
